bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/FindReplaceLogic.java

```diff
@@ -542,12 +543,11 @@ private Point replaceSelection(String replaceString, boolean regExReplace) {
 	 * options.
 	 *
 	 * @param findString             the string to search for
-	 * @param forwardSearch          the direction of the search
 	 * @return <code>true</code> if the search string can be found using the given
 	 *         options
 	 *
 	 */
-	private boolean findNext(String findString, boolean forwardSearch) {
+	private boolean findNext(String findString) {
```

Comment:
Just for my understanding: is this a kind of cleanup or is there a functional reason for this change?

Reply:
The historic findNext method was passed these arguments from the Dialog. This method is only used internally (and is private) and was always used like this: findNext(-, isActive(forwardSearch)). Additionally, the logic part of the method (findNext (string, position)) already discarded this parameter, instead checking for internal state.
The only reason this parameter was used was for generating the right search offset position.

This has a few problems:
- the method signature is conflated to require more parameters than it actually should
- it is possible to generate broken behaviour by calling the method as such: `findNext(findString, !isActive(SearchOptions.FORWARD))`

Functionally, everything stays the same, and since the only reference to this method called the method with an argument polled from internal state anyway, I believe this is a big improvement on the status quo. I do not intend (and could not find) any functional changes

Reply:
Thank you for the explanation! That it totally fine. I justed wanted to know whether I have to consider any behavioral change because of this when reviewing the change or whether it is only a refactoring.